pkg/descheduler/framework/plugins/loadaware/utilization_util.go

```diff
@@ -243,40 +312,115 @@ func evictPodsFromSourceNodes(
 	continueEviction continueEvictionCond,
 	evictionReasonGenerator evictionReasonGeneratorFn,
 ) {
-	var targetNodes []*corev1.Node
-	totalAvailableUsages := map[corev1.ResourceName]*resource.Quantity{}
-	for _, destinationNode := range destinationNodes {
-		targetNodes = append(targetNodes, destinationNode.node)
+	totalAvailableUsages, targetNodes := targetAvailableUsage(destinationNodes, resourceNames, false)
+	prodAvailableUsages, prodTargetNodes := targetAvailableUsage(prodDestinationNodes, resourceNames, true)
+	bothTotalAvailableUsage, bothTotalNodes := targetAvailableUsage(bothDestinationNodes, resourceNames, false)
+	prodBothAvailableUsage, prodBothTotalNodes := targetAvailableUsage(bothDestinationNodes, resourceNames, true)
+	klog.V(4).InfoS("node pool availableUsage", "onlyNodeTotal", totalAvailableUsages, "onlyProdOnly", prodAvailableUsages,
+		"bothLowNodesTotal", bothTotalAvailableUsage, "bothLowProdTotal", prodBothAvailableUsage)
+
+	nodeTotalAvailableUsages := newAvailableUsage(resourceNames)
+	for _, resourceName := range resourceNames {
+		if quantity, ok := nodeTotalAvailableUsages[resourceName]; ok {
+			if _, totalOk := totalAvailableUsages[resourceName]; totalOk {
+				quantity.Add(*totalAvailableUsages[resourceName])
+			}
+			if _, prodOk := bothTotalAvailableUsage[resourceName]; prodOk {
+				quantity.Add(*bothTotalAvailableUsage[resourceName])
+			}
+		}
+	}
+	nodeKeysAndValues := []interface{}{
+		"nodePool", nodePoolName,
+	}
+	for resourceName, quantity := range nodeTotalAvailableUsages {
+		nodeKeysAndValues = append(nodeKeysAndValues, string(resourceName), quantity.String())
+	}
+	klog.V(4).InfoS("Total node usage capacity to be moved", nodeKeysAndValues...)
 
-		for _, resourceName := range resourceNames {
-			quantity, ok := totalAvailableUsages[resourceName]
-			if !ok {
-				switch resourceName {
-				case corev1.ResourceCPU:
-					quantity = resource.NewMilliQuantity(0, resource.DecimalSI)
-				case corev1.ResourceMemory, corev1.ResourceEphemeralStorage, corev1.ResourceStorage:
-					quantity = resource.NewQuantity(0, resource.BinarySI)
-				default:
-					quantity = resource.NewQuantity(0, resource.DecimalSI)
-				}
-				totalAvailableUsages[resourceName] = quantity
+	targetNodes = append(targetNodes, bothTotalNodes...)
+	balancePods(ctx, nodePoolName, sourceNodes, targetNodes, nodeTotalAvailableUsages, dryRun, nodeFit, false, resourceWeights, podEvictor,
+		podFilter, nodeIndexer, continueEviction, evictionReasonGenerator)
+
+	// bothLowNode will be used by nodeHigh and prodHigh nodes, needs sub resources used by pods on nodeHigh.
+	for _, resourceName := range resourceNames {
+		if quantity, ok := nodeTotalAvailableUsages[resourceName]; ok {
+			if bothTotalAvailableUsage[resourceName].Cmp(*quantity) > 1 {
+				bothTotalAvailableUsage[resourceName].Sub(*quantity)
+			} else {
+				// all resource used by nodeHigh pods
+				bothTotalAvailableUsage[resourceName].Sub(*bothTotalAvailableUsage[resourceName])
```

Comment:
Is the logic here right? I think if bothTotalAvialbleUsage is bigger than nodeTotalAvailableUsage, it means that bothTotalAvailableUsage is not used. So nothing need to do here. This makes it Zero.